corehq/apps/data_interfaces/utils.py

```diff
@@ -34,3 +35,33 @@ def add_cases_to_case_group(domain, case_group_id, uploaded_data):
         case_group.save()
 
     return response
+
+
+def archive_forms(user, uploaded_data):
```

Comment:
I think if you can find a good filesize limit that's a reasonable proxy for # of forms then that makes sense. if not, then probably just reading the rows twice is not going to be too horrible